lms/static/scripts/frontend_apps/components/AutoGradingConfigurator.tsx

```diff
@@ -0,0 +1,208 @@
+import {
+  Checkbox,
+  CheckboxCheckedFilledIcon,
+  Input,
+  RadioGroup,
+} from '@hypothesis/frontend-shared';
+import type { ComponentChildren } from 'preact';
+import { useCallback, useId } from 'preact/hooks';
+
+export type GradingType = 'all_or_nothing' | 'scaled';
+
+export type AutoGradingConfig = {
+  /** Whether auto grading is active for the assignment or not */
+  active?: boolean;
+
+  /**
+   * - all_or_nothing: students need to meet a minimum value, making them get
+   *                   either 0% or 100%
+   * - scaled: students may get a proportional grade based on the amount of
+   *           annotations. If requirement is 4, and they created 3, they'll
+   *           get a 75%
+   */
+  gradingType: GradingType;
+
+  /**
+   * - cumulative: both annotations and replies will be counted together for
+   *               the grade calculation
+   * - separately: students will have different annotation and reply goals.
+   */
+  activityCalculation: 'cumulative' | 'separately';
+
+  /**
+   * Required number of annotations if activityCalculation is 'separately' or
+   * annotations+replies otherwise
+   */
+  requiredAnnotations: number;
+
+  /**
+   * Required number of replies if activityCalculation is 'separately'
+   */
+  requiredReplies?: number;
+};
+
+type AnnotationsGoalInputGroupProps = {
+  children?: ComponentChildren;
+  gradingType: GradingType;
+  value: number;
+  onChange: (newValue: number) => void;
+};
+
+/**
+ * Controls containing a number input to set the amount of required annotations
+ * or replies
+ */
+function AnnotationsGoalInputGroup({
+  children,
+  gradingType,
+  value,
+  onChange,
+}: AnnotationsGoalInputGroupProps) {
+  const inputId = useId();
+
+  return (
+    <div className="flex gap-2 items-center">
+      <label
+        className="grow flex justify-between items-center"
+        htmlFor={inputId}
+      >
+        {children}
+        <span className="uppercase font-semibold">
+          {gradingType === 'all_or_nothing' ? 'Minimum' : 'Goal'}
+        </span>
+      </label>
+      <Input
+        id={inputId}
+        classes="max-w-14"
+        type="number"
+        required
+        min={1}
```

Comment:
Should this be 0? In case an instructor want to ask for annos but not replies (or vice versa)?

Not sure.

Reply:
That's actually a good point. I think the right way to validate this is to make sure at least one is greater than zero, but we should allow one of them to be zero.

Let me give it a thought.

Reply:
It is simpler to validate the fields if we can do so individually, so it might be sufficient to require 1 annotation and 0 replies. Ask around internally and see what folks think?

Reply:
Yeah, I think that would be the case almost always. I guess there's a chance that some teacher wants to require a specific amount of replies and no annotations, but we could change that later if it really happens.